src/apps/dfx/ethereum/dfx.amm.token-fetcher.ts

```diff
@@ -0,0 +1,103 @@
+import { getAppImg } from '~app-toolkit/helpers/presentation/image.present';
+import { ContractType } from '~position/contract.interface';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { AppTokenPosition } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+
+import { Addresses } from '../addresses';
+import { DfxContractFactory } from '../contracts';
+import { DFX_DEFINITION } from '../dfx.definition';
+
+const appId = DFX_DEFINITION.id;
+const groupId = DFX_DEFINITION.groups.curve.id;
+const network = Network.ETHEREUM_MAINNET;
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class EthereumDfxCurveTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(DfxContractFactory) private readonly dfxContractFactory: DfxContractFactory,
+  ) {}
+
+  async getPositions() {
+    const curveAddresses = Addresses[network].amm.map(v => v.curve);
+
+    const baseTokenDependencies = await this.appToolkit.getBaseTokenPrices(network);
+
+    // Create multipcall wrapper for batch RPC calls
+    const multicall = this.appToolkit.getMulticall(network);
+
+    const lpTokens = await Promise.all(
+      curveAddresses.map(async curveAddress => {
+        // Fetch data from DFX curve contract
+        const contract = this.dfxContractFactory.dfxCurve({ address: curveAddress, network });
+        const [name, symbol, decimals, supplyRaw, token0AddressRaw, token1AddressRaw, [totalLiquidityRaw]] =
+          await Promise.all([
+            multicall.wrap(contract).name(),
+            multicall.wrap(contract).symbol(),
+            multicall.wrap(contract).decimals(),
+            multicall.wrap(contract).totalSupply(),
+            multicall.wrap(contract).numeraires(0),
+            multicall.wrap(contract).numeraires(1),
+            multicall.wrap(contract).liquidity(),
+          ]);
+        const underlyingTokenAddresses = [token0AddressRaw.toLowerCase(), token1AddressRaw.toLowerCase()];
+
+        // Find underlying tokens from base dependencies
+        const [token0, token1] = underlyingTokenAddresses.map(tokenAddress =>
+          baseTokenDependencies.find(v => v.address === tokenAddress.toLowerCase()),
+        );
+
+        // Remove any pools containing tokens that aren't available
+        if (!token0 || !token1) return null;
+        const tokens = [token0, token1];
+
+        // Denormalize big integer values
+        const supply = Number(supplyRaw) / 10 ** decimals;
+        const totalLiquidity = Number(totalLiquidityRaw) / 1e18;
+        const pricePerShare = 1;
+        const price = totalLiquidity / supply;
+
+        // Prepare display props
+        const [, baseToken, quoteToken] = name.split('-');
+        const label = `${baseToken.toUpperCase()}/${quoteToken.toUpperCase()} LP`;
```

Comment:
For Uniswap V2, we set the label as `UNI-V2 WETH / WBTC`. Perhaps you should label DFX as:

```ts
const label = `DFX LP ${baseToken.toUpperCase()} / ${quoteToken.toUpperCase()};`
```

Reply:
Fix https://github.com/Zapper-fi/studio/pull/347/commits/024456bdc2064f0701bbcfaf0576aa49dee26c22